workers/tests/unittests/test_ohsome_client.py

```diff
@@ -40,15 +40,31 @@ def setUp(self) -> None:
 
     def test_query_valid_response(self) -> None:
         with patch("httpx.AsyncClient.post", new_callable=AsyncMock) as mock_request:
-            mock_request.return_value = httpx.Response(200, content=self.valid_geojson)
+            mock_request.return_value = httpx.Response(
+                200,
+                content=self.valid_geojson,
+                request=httpx.Request("POST", "mock.org"),
+            )
             response = asyncio.run(ohsome_client.query(self.layer, self.bpolys))
             self.assertTrue(response.is_valid)
 
     def test_query_invalid_response_with_status_code_200(self) -> None:
         """When response is streamed it can be invalid while status code equals 200"""
         with patch("httpx.AsyncClient.post", new_callable=AsyncMock) as mock_request:
             mock_request.return_value = httpx.Response(
-                200, content=self.invalid_geojson
+                200,
+                content=self.invalid_geojson,
+                request=httpx.Request("POST", "mock.org"),
             )
-            response = asyncio.run(ohsome_client.query(self.layer, self.bpolys))
-            self.assertIsNone(response)
+            with self.assertRaises(JSONDecodeError):
+                asyncio.run(ohsome_client.query(self.layer, self.bpolys))
+
+    def test_query_status_code_400(self) -> None:
+        with patch("httpx.AsyncClient.post", new_callable=AsyncMock) as mock_request:
+            mock_request.return_value = httpx.Response(
+                400,
+                content=self.invalid_geojson,
+                request=httpx.Request("POST", "mock.org"),
+            )
+            with self.assertRaises(httpx.HTTPStatusError):
+                asyncio.run(ohsome_client.query(self.layer, self.bpolys))
```

Comment:
I am not sure if we want to tests the function like this because the value of the first part of the url (https://api.ohsome.org/v1 in this case) is depended on an evironment variable (OHSOME_API). But I guess we can simple read the env and use it.

Reply:
I took the ohsome api url as definied in definitions.py for the tests.